python/az/aro/azext_aro/custom.py

```diff
@@ -202,14 +206,18 @@ def validate(cmd,  # pylint: disable=too-many-locals,too-many-statements
              warnings_as_text=False):
 
     class mockoc:  # pylint: disable=too-few-public-methods
-        def __init__(self, disk_encryption_id, master_subnet_id, worker_subnet_id):
+        def __init__(self, disk_encryption_id, master_subnet_id, worker_subnet_id, preconfigured_nsg):
+            self.network_profile = openshiftcluster.NetworkProfile(
+                preconfigured_nsg='Enabled' if preconfigured_nsg else 'Disabled'
+            )
             self.master_profile = openshiftcluster.MasterProfile(
                 subnet_id=master_subnet_id,
                 disk_encryption_set_id=disk_encryption_id
             )
             self.worker_profiles = [openshiftcluster.WorkerProfile(
                 subnet_id=worker_subnet_id
             )]
+            self.worker_profiles_status = None
```

Comment:
q: I'm sure this is fine, but it's unintuitive at a glace and now I'm curious: why was this needed?

Reply:
Not Setting/defining the field worker_profiles_status here will result in exception when this function makes a call to get_cluster_network_resources() since it loops through the worker_profiles_status. During az create we don't pass any values to this filed as it's readOnly.

Reply:
IMO the RP code should be able to handle the case of worker_profiles_status being nil?